src/session/session_compact.c

```diff
@@ -360,6 +360,31 @@ __compact_worker(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __wt_compact_check_eligibility --
+ *     Function to check whether the specified URI is eligible for compaction.
+ */
+int
+__wt_compact_check_eligibility(WT_SESSION_IMPL *session, const char *uri)
+{
+    bool is_server;
+
+    is_server = session == S2C(session)->background_compact.session;
+
+    /* The background compaction server should not compact the history store file. */
+    if (is_server && WT_STREQ(uri, WT_HS_URI))
+        return (ENOTSUP);
+
+    /*
+     * Tiered tables cannot be compacted. Don't print any message if this function is executed by
+     * the background compaction server as it is handled internally.
+     */
+    if (WT_SUFFIX_MATCH(uri, ".wtobj"))
+        WT_RET(is_server ? ENOTSUP : __wt_object_unsupported(session, uri));
```

Comment:
Based on the above comment, we do not need to check for the background server, just check and return ENOTSUP error for the not eligible files and in the foreground code flow, the `__wt_object_unsupported` error can be raised.

Reply:
Thanks for the suggestion, I agree it makes the code better, see 4dbee735c39da2b996de7ac82e5b59fe562bc66e.